library/core/src/time.rs

```diff
@@ -486,6 +522,8 @@ impl Duration {
     #[rustc_const_stable(feature = "duration_consts", since = "1.32.0")]
     #[must_use]
     #[inline]
+    #[requires(self.is_safe())]
```

Comment:
For safe functions, you should be able to assume that is the case. So this becomes redundant.